### DISCUSSION TITLE
What do you think about my Openstreetmap extension?

### DISCUSSION THREAD

**Discussion by GiovanniSalmeri** [datenstrom/community#408] [Ask a question] [answered]
All good things come in three's: after the official extension [Googlemaps](https://github.com/datenstrom/yellow-extensions/tree/master/features/googlemaps) and the experimental one [Leaflet](https://github.com/nibreh/yellow-plugin-leaflet) by @nibreh I have written another [little extension for embedding OpenStreetMap maps](https://github.com/GiovanniSalmeri/yellow-openstreetmap). It is different from the latter (which also uses OSM) in that:

1. it has *not* collaborative features
2. it has an interface as similar as possible to the Googlemaps extension
3. it allows to choose between the different layers of OSM
4. it allows to set very easily the marker
5. it doesn't use a third-party JS library (like Leaflet) but only the native interface of OSM (this doesn't mean that it works if JS is disabled!)

I haven't done extensive tests yet, but the basics should be working. Any comment is appreciated!

**Comment by GiovanniSalmeri** [comment, marked as answer]
Now added support for textual addresses, via Nominatim geolocation. With this addition the extension should be a drop-in replacement for Googlemaps.

**Comment by GiovanniSalmeri** [comment]
A little update: the latest version of [Openstreetmap](https://github.com/GiovanniSalmeri/yellow-openstreetmap) now allows multiple points of interest (possibly with a popup with name and/or description). For this functionality, the library [Leaflet](https://leafletjs.com/) is used, but when points of interest are not needed the same ultra-light and ultra-fast mechanism as before is used instead (and Leaflet is not even loaded).

As regards points of interest: after devising a simple CSV format for describing them, I realised that it was almost identical with the Garmin CSV format (that I didn't even know). So I have done a little modification and now it is identical. This means that any Garmin POI file can be used (in the web there are tons of them). The format is very easy also to handwrite, and this extension accept moreover textual addresses (even easier to write!).

**Reply by annaesvensson** [reply]
Tip: Use a self hosted version of Leaflet instead of a CDN hosted version of Leaflet. 

Since Datenstrom Yellow 0.8 most extensions follow this software design pattern, whenever possible extensions should come with all files needed, including JS, CSS and images. There are benefits for self hosting, such as reduced reliance on third-party providers, fewer tracking cookies and improved data protection compliance, GDPR for Europeans. Of course, you can load large resources on demand or make use of the static bundler pipeline. You do you.

**Reply by GiovanniSalmeri** [reply]
Thank you for the tip, done with https://github.com/GiovanniSalmeri/yellow-openstreetmap/commit/f6025efcd17dba6d044bb3fadaebfaf372e0a67a
